scripts/release-notes.py

```diff
@@ -1,67 +1,121 @@
 #
 # Requires personal GitHub token with default permissions:
-#   https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token
+#   https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
 #
-# Usage: python release-notes.py <github_token> <jaeger-repo> <num-commits>
+# Usage: python release-notes.py --help
 #
 
+import argparse
 import json
+import urllib.parse
+import os
+import sys
 from urllib.request import (
     urlopen,
     Request
 )
-import urllib.parse
-import sys
+
 
 def eprint(*args, **kwargs):
     print(*args, file=sys.stderr, **kwargs)
 
-if len(sys.argv) < 4:
-    eprint("Usage: python release-notes.py <github_token> <jaeger-repo> <num-commits>")
-    sys.exit(1)
-
-token=sys.argv[1]
-repo=sys.argv[2]
-num_commits=sys.argv[3]
-
-accept_header="application/vnd.github.groot-preview+json"
-commits_url='https://api.github.com/repos/jaegertracing/{repo}/commits'.format(repo=repo)
-pulls_url="{commits_url}/{commit_sha}/pulls"
-
-# load commits
-data = urllib.parse.urlencode({'per_page': num_commits})
-req=Request(commits_url + '?' + data)
-print(req.full_url)
-req.add_header('Authorization', 'token {token}'.format(token=token))
-commits=json.loads(urlopen(req).read())
-print('Retrieved', len(commits), 'commits')
-
-# load PR for each commit and print summary
-for commit in commits:
-    sha = commit['sha']
-    msg_lines = commit['commit']['message'].split('\n')
-    msg = msg_lines[0]
-    req = Request(pulls_url.format(commits_url=commits_url,commit_sha=sha))
-    req.add_header('accept', accept_header)
-    req.add_header('Authorization', 'token {token}'.format(token=token))
-    pulls = json.loads(urlopen(req).read())
-    if len(pulls) > 1:
-        print("More than one pull request for commit {commit}".format(commit=sha))
-    # TODO handle commits without pull requests
-    pull = pulls[0]
-    pull_no = pull['number']
-    pull_url = pull['html_url']
-    author = pull['user']['login']
-    author_url = pull['user']['html_url']
-    msg = msg.replace('(#{pull})'.format(pull=pull_no), '').strip()
-    print('* {msg} ([@{author}]({author_url}) in [#{pull}]({pull_url}))'.format(
-        msg=msg,
-        author=author,
-        author_url=author_url,
-        pull=pull_no,
-        pull_url=pull_url,
-    ))
+
+def num_commits_before_prev_tag(token, base_url):
+    tags_url = f"{base_url}/tags"
+
+    req = Request(tags_url)
+    req.add_header("Authorization", f"token {token}")
+    tags = json.loads(urlopen(req).read())
+    prev_release_tag = tags[0]['name']
+    compare_url = f"{base_url}/compare/master...{prev_release_tag}"
+    req = Request(compare_url)
+    req.add_header("Authorization", f"token {token}")
+    compare_results = json.loads(urlopen(req).read())
+    num_commits = compare_results['behind_by']
+
+    print(f"master is behind {prev_release_tag} by {num_commits} commits")
```

Comment:
"behind" sounds like you need a pull. I'd rather say "N new commits since prevtag "